src/upgrades.ts

```diff
@@ -74,31 +73,44 @@ export async function deployProxy<T extends Contract>(
 
   // Let the transaction propagate across the ethereum nodes. This is mostly to
   // wait for all Alchemy nodes to catch up their state.
-  await contractInstance.deployTransaction.wait(1)
+  await contractInstance.deploymentTransaction()?.wait(1)
+
+  const contractAddress = await contractInstance.getAddress()
 
   log(
-    `Deployed ${name} as ${opts?.proxyOpts?.kind || "transparent"} proxy at ${
-      contractInstance.address
-    } (tx: ${contractInstance.deployTransaction.hash})`
+    `Deployed ${name} as ${
+      opts?.proxyOpts?.kind || "transparent"
+    } proxy at ${contractAddress} (tx: ${
+      contractInstance.deploymentTransaction()?.hash
+    })`
   )
 
   const artifact = artifacts.readArtifactSync(opts?.contractName || name)
 
-  const adminInstance = await upgrades.admin.getInstance()
-  const implementation = await adminInstance.getProxyImplementation(
-    contractInstance.address
+  const implementation = await upgrades.erc1967.getImplementationAddress(
+    contractAddress
   )
 
-  const transactionReceipt = await ethers.provider.getTransactionReceipt(
-    contractInstance.deployTransaction.hash
-  )
+  const transactionHash = contractInstance.deploymentTransaction()?.hash
+
+  const transactionReceipt = transactionHash
+    ? await hre.ethers.provider.getTransactionReceipt(transactionHash)
+    : null
+
+  if (!transactionReceipt) {
+    throw new Error(
+      `Could not find transaction receipt for transaction hash: ${transactionHash}`
+    )
+  }
 
   const deployment: Deployment = {
-    address: contractInstance.address,
+    address: contractAddress,
     abi: artifact.abi,
-    transactionHash: contractInstance.deployTransaction.hash,
+    transactionHash,
     implementation: implementation,
+    // @ts-ignore-next-line 
     receipt: transactionReceipt,
+    // @ts-ignore-next-line 
     libraries: opts?.factoryOpts?.libraries,
```

Comment:
`libraries` fixed by adjusting the type 8eece55

`receipt` - no idea how to solve it for now as these types are incompatible, removed the `receipt` for now as this is an optional field on `Deployment` type. Not sure if we can leave it like that, end user should be able to get receipt if needed anyway because there is `transactionHash` saved in the deployment 7adb77f

Reply:
What we wanted to achieve was to output deployment artifacts matching the output of the [`hardhat-helpers`](https://github.com/wighawag/hardhat-deploy) plugin. Unfortunately `hardhat-helpers` are still based on ethers v5. I'm fine with the temporary removal of the `receipt` property.